src/main/java/org/sonarlint/intellij/promotion/PromotionProvider.kt

```diff
@@ -112,15 +113,18 @@ class PromotionProvider(private val project: Project) {
         val wasReportEverUsed = PropertiesComponent.getInstance().getBoolean(WAS_REPORT_EVER_USED, false)
 
         if (!wasReportEverUsed && firstAutoAnalysis.isBefore(Instant.now().minus(FULL_PROJECT_PROMOTION_DURATION))) {
-            showPromotion(notifications, "Detect issues in your whole project")
+            UiUtils.Companion.runOnUiThread(project, Runnable {
+                SonarLintUtils.getService(project, PromotionProvider::class.java)
+                    .showPromotion(notifications, "Detect issues in your whole project")
+            })
```

Comment:
You can simplify everything to just
```
            UiUtils.runOnUiThread(project) {
                showPromotion(notifications, "Detect issues in your whole project")
            }
```
And then, I don't necessarily understand what is the need for this to run of the UI thread, what was the issue?

Reply:
There are 2 different notifications can be triggered at the same time from multiple threads and we might end up showing 2 notifications which is something we want to avoid(We have 7 days rule)